community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/conf.py

```diff
@@ -440,10 +440,53 @@ def render_conf_lines(self) -> Iterable[str]:
         for s in sorted(self.switches.values(), key=lambda s: s.name):
             yield from s.render_conf_lines()
 
+class TopologySummary:
+    """
+    Represents a summary of the topology, to make judgements about changes.
+    To be stored in JSON file along topology.conf to simplify parsing.
+    """
+    def __init__(
+            self,
+            physical_host: Optional[Dict[str, str]] = None,
+            down_nodes: Optional[Iterable[str]] = None,
+            tpu_nodes: Optional[Iterable[str]] = None,
+        ) -> None:
+        self.physical_host = physical_host or {}
+        self.down_nodes = set(down_nodes or [])
+        self.tpu_nodes = set(tpu_nodes or [])
+
+
+    @classmethod
+    def loads(cls, s: str) -> "TopologySummary":
+        d = json.loads(s)
+        return cls(
+            physical_host=d.get("physical_host"),
+            down_nodes=d.get("down_nodes"),
+            tpu_nodes=d.get("tpu_nodes"),
+        )
+    
+    def dumps(self) -> str:
+        return json.dumps(
+            {
+                "physical_host": self.physical_host,
+                "down_nodes": list(self.down_nodes),
+                "tpu_nodes": list(self.tpu_nodes),
+            },
+            indent=2)
+    
+    def _nodenames(self) -> Set[str]:
+        return set(self.physical_host) | self.down_nodes | self.tpu_nodes
+        
+    def requires_reconfigure(self, prev: "TopologySummary") -> bool:
+        """
+        Reconfigure IFF the nodes were added
+        """
+        return len(self._nodenames() - prev._nodenames()) > 0
```

Comment:
What if a VM changed physical host? Such as in a maintenance event?

Reply:
In description:
> The logic of requires_reconfigure is very simple, to be made more complex in follow up PRs.